docs/08_getting_started.md

```diff
@@ -0,0 +1,329 @@
+
+        for (Map.Entry<String, List<String>> pair : headers.entrySet()) {
+            final String headerName = pair.getKey();
+            final List<String> headerValueList = pair.getValue();
+
+            if (headerValueList.size() > 1) {
+                requestBuilder.removeHeader(headerName);
+                for (String headerValue : headerValueList) {
+                    requestBuilder.addHeader(headerName, headerValue);
+                }
+            } else if (headerValueList.size() == 1) {
+                requestBuilder.header(headerName, headerValueList.get(0));
+            }
+
+        }
+
+        final okhttp3.Response response = okhttp.newCall(requestBuilder.build()).execute();
+
+        if (response.code() == 429) {
+            response.close();
+
+            throw new ReCaptchaException("reCaptcha Challenge requested", url);
+        }
+
+        final ResponseBody body = response.body();
+        String responseBodyToReturn = null;
+
+        if (body != null) {
+            responseBodyToReturn = body.string();
+        }
+
+        final String latestUrl = response.request().url().toString();
+        return new Response(response.code(), response.message(), response.headers().toMultimap(),
+                responseBodyToReturn, latestUrl);
+    }
+}
+
+
+```
+
+</details>
+
+
+You can find two example implementations of the Downloader below.
+Both use an old OkHttp version (version 3) to support Android 4. However, you do not need to use OkHttp.
+In case you want to use it and do not need to support old Android versions,
+we recommend using a newer OkHttp version, because the version used in the example will only receive security updates until December 2021.
+
+- [Downloader implementation for NewPipe Extractor tests](https://github.com/TeamNewPipe/NewPipeExtractor/blob/master/extractor/src/test/java/org/schabi/newpipe/downloader/DownloaderTestImpl.java)
+- [Downloader implementation of the NewPipe app](https://github.com/TeamNewPipe/NewPipe/blob/master/app/src/main/java/org/schabi/newpipe/DownloaderImpl.java)  
+Note: This implementation contains some additional methods which are not necessary for general use cases.
+
+
+Depending on your use case and your downloader implementation, we suggest you to make use of the singleton pattern.
+
+#### Browser Agent
+
+When making requests to websites, you should define a user agent for the request headers. We have build our tests on
+```
+Mozilla/5.0 (Windows NT 10.0; rv:78.0) Gecko/20100101 Firefox/78.0
+```
+For this reason, we strongly recommend using that agent or any other that is used in the [testimplementation in NewPipe Extractor](https://github.com/TeamNewPipe/NewPipeExtractor/blob/master/extractor/src/test/java/org/schabi/newpipe/downloader/DownloaderTestImpl.java).  
+However, a service might want to override the default user agent.
+This is sometimes necessary to get special versions of a website (e.g. mobile or legacy version).
```

Comment:
Do you have a specific example of when it is needed? I know there are some cases where it is need inside the extractor, but I don't think someone using the extractor (outside of it) needs to concern themselves with that.